### PR TITLE
Finalize News List

### PR DESCRIPTION
**Pull request by wiseodd**
Now everything works on news list:
1. Infinite scroll
2. Change news category, by selecting it in sidenav

### CODE REVIEW

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java [hunk 1]
```diff
@@ -126,9 +125,78 @@ public void onError(Throwable e) {
...
+        }
+
+        return ret.cache();
+    }
+
+    private void setUpNewsList(View view) {
+        newsList = (RecyclerView) view.findViewById(R.id.news_list);
+
+        final LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
+        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
```
Comment (timotiusnc) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java:171]: is this to force the layout orientation to always be vertical (not landscape)?

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java [hunk 2]
```diff
@@ -126,9 +125,78 @@ public void onError(Throwable e) {
...
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
+
+                LinearLayoutManager lm = (LinearLayoutManager) recyclerView.getLayoutManager();
+
+                int visibleItemCount = recyclerView.getChildCount();
+                int totalItemCount = lm.getItemCount();
+                int firstVisibleItem = lm.findFirstVisibleItemPosition();
+
+                if (!isLoading && (totalItemCount - visibleItemCount) <= (firstVisibleItem + LOADING_THRESHOLD)) {
```
Comment (timotiusnc) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java:192]: What does `totalItemCount - visibleItemCount` and `firstVisibleItem + LOADING_THRESHOLD` mean? And what does the comparison (`<=`) between them represent?

**Comment by timotiusnc** [comment]
It seems that we finally run into problem on retaining activity/fragment on device configuration change (in this case: orientation change).

When I change the orientation, the entire list is refreshed and user is presented with News from the beginning although they have scrolled to next page.

Maybe this [sample from RxAndroid](https://github.com/ReactiveX/RxAndroid/blob/60741117c936b198ebc89dcc058ccaaa2b09ebfb/sample-app/src/main/java/rx/android/samples/ListFragmentActivity.java) repo could help.

Another resource: [this](http://stackoverflow.com/questions/15313598/once-for-all-how-to-correctly-save-instance-state-of-fragments-in-back-stack) or [this](http://stackoverflow.com/questions/11182180/understanding-fragments-setretaininstanceboolean) SO posts. Or maybe [this blogpost](http://www.androiddesignpatterns.com/2013/04/retaining-objects-across-config-changes.html).

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java (hunk 1 above)
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java:171]: It's different than what you think.

Vertical - Horizontal is the flow of child widget in the layout.
Portrait - Landscape is the orientation of the screen.

**Comment by wiseodd** [comment]
Done. Please check again.

I'm using Occam's Razor here, in layman term: "If two things yield the same result, the simpler one is the better".

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java (hunk 2 above)
Comment (timotiusnc) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java:192]: `totalItemCount` = total in the list
`visibleItemCount` = total item that is currently visible in the device
`firstVisibleItem` = index of first item that is visibile in the device

Basically the expression is used to determine whether the app should display loading icon or not.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -8,28 +8,26 @@
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:theme="@style/PumpkinTheme" >
+        android:theme="@style/PumpkinTheme">
         <activity
             android:name=".MainActivity"
-            android:label="@string/app_name" >
+            android:configChanges="orientation|screenSize"
```
Comment (timotiusnc) [inline app/src/main/AndroidManifest.xml:14]: wow, apparently the solution needed is just this one line of code!
